src/app/store-stats/CharacterStats.tsx

```diff
@@ -2,161 +2,105 @@ import React from 'react';
-        }
-        return next + cooldown;
-      }
-    });
+function CharacterStat({ stats, storeId, className }: CharacterStatProps) {
+  return (
+    <div className={clsx('stat-row', className)}>
+      {stats.map(({ stat, tooltip }) => {
+        // if this is the "max gear power" stat (hash -3),
+        // add in an onClick and an extra class
+        const isMaxGearPower = stat.hash === -3 && storeId;
 
-    return (
-      <div className="stat-bars">
-        {statList.map((stat, index) => (
-          <PressTip key={stat.hash} tooltip={tooltips[index]}>
-            <div className="stat">
+        return (
+          <PressTip key={stat.hash} tooltip={tooltip} allowClickThrough={true}>
+            <div
+              className={clsx('stat', { pointerCursor: isMaxGearPower })}
+              aria-label={`${stat.name} ${stat.value}`}
+              role={isMaxGearPower ? 'button' : 'group'}
+              onClick={
+                isMaxGearPower
+                  ? () => {
+                      showGearPower(storeId!);
+                    }
+                  : undefined
+              }
+            >
               <img src={stat.icon} alt={stat.name} />
-              {getD1CharacterStatTiers(stat).map((n, index) => (
-                <div key={index} className="bar">
-                  <div
-                    className={clsx('progress', {
-                      complete: n / 60 === 1,
-                    })}
-                    style={{ width: percent(n / 60) }}
-                  />
-                </div>
-              ))}
+              <div>
+                {stat.hash < 0 ? (
+                  <span className="powerStat">
+                    <FractionalPowerLevel power={stat.value} />
+                  </span>
+                ) : (
+                  stat.value
+                )}
+                {(stat.hasClassified || stat.differentEquippableMaxGearPower) && (
+                  <sup className="asterisk">*</sup>
+                )}
+              </div>
             </div>
           </PressTip>
-        ))}
-      </div>
-    );
-  } else {
-    const powerTooltip = (stat: DimCharacterStat): React.ReactNode => (
-      <>
-        {`${stat.name}${stat.hasClassified ? `\n\n${t('Loadouts.Classified')}` : ''}`}
-        {stat.richTooltip && (
-          <>
-            <hr />
-            <div className="richTooltipWrapper">
-              {stat.richTooltip}
-              {stat.differentEquippableMaxGearPower && (
-                <div className="tooltipFootnote">* {t('General.ClickForDetails')}</div>
-              )}
-            </div>
-          </>
-        )}
-      </>
-    );
-    const powerInfos: {
-      stat: DimCharacterStat;
-      tooltip: React.ReactNode;
-    }[] = _.compact([stats.maxTotalPower, stats.maxGearPower, stats.powerModifier]).map((stat) => ({
+        );
+      })}
+    </div>
+  );
+}
+
+export const PowerFormula = React.memo(function ({ stats, storeId }: Props) {
```

Comment:
I generally avoid React.memo unless I think it'll really get rendered repeatedly with the same arguments. Most of the time (especially with react-redux `connect`) things will only be re-rendering when the inputs change anyway, so `memo` just adds overhead.

Reply:
ah ok updated, previously just copied from existing logic from https://github.com/DestinyItemManager/DIM/blob/master/src/app/store-stats/CharacterStats.tsx#L162